src/VisualStudio/Core/Def/Indentation/VisualStudioInferredIndentationService.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Indentation;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.Text.Editor;
+
+namespace Microsoft.VisualStudio.LanguageServices.Indentation
+{
+    [ExportWorkspaceService(typeof(IInferredIndentationService), ServiceLayer.Host), Shared]
+    internal sealed class VisualStudioInferredIndentationService : IInferredIndentationService
+    {
+        private readonly IIndentationManagerService _indentationManagerService;
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public VisualStudioInferredIndentationService(IIndentationManagerService indentationManagerService)
+        {
+            _indentationManagerService = indentationManagerService;
+        }
+
+        public async Task<DocumentOptionSet> GetDocumentOptionsWithInferredIndentationAsync(Document document, bool explicitFormat, CancellationToken cancellationToken)
```

Comment:
This method is a copy/paste of the deleted version in src/EditorFeatures/Core/Shared/Extensions/InferredIndentationExtensions.cs, removing the `IIndentationManagerService` parameter and adding a `_` to access the class field instead.

Reply:
This part LGTM.